File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/location/LocationRepository.java
package com.a65apps.kalimullinilnazrafilovich.interactors.location;

import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo;
import com.a65apps.kalimullinilnazrafilovich.entities.Location;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

public interface LocationRepository {
    void insertContactLocation(Location location, ContactDetailsInfo contactDetailsInfo);

    Single<List<Location>> getAllContactLocations();

    Single<Location> createOrUpdateContactLocationByCoordinate(ContactDetailsInfo contactDetailsInfo,
                                                               double latitude, double longitude);
}

File: java/src/main/java/com/a65apps/kalimullinilnazrafilovich/interactors/location/LocationRepository.kt
package com.a65apps.kalimullinilnazrafilovich.interactors.location

import com.a65apps.kalimullinilnazrafilovich.entities.ContactDetailsInfo
import com.a65apps.kalimullinilnazrafilovich.entities.Location
import io.reactivex.rxjava3.core.Single
import kotlinx.coroutines.flow.Flow

interface LocationRepository {
    fun insertContactLocation(location: Location?, contactDetailsInfo: ContactDetailsInfo?)

    fun getAllContactLocations(): Flow<List<Location>>?

    fun createOrUpdateContactLocationByCoordinate(contactDetailsInfo: ContactDetailsInfo?,
                                                  latitude: Double, longitude: Double): Single<Location?>?
}